xwayland/xwm.c

```diff
@@ -238,13 +239,20 @@ static void xwm_set_net_client_list(struct wlr_xwm *xwm) {
 			XCB_ATOM_WINDOW, 32, mapped_surfaces, windows);
 }
 
+static void xsurface_set_net_wm_state(struct wlr_xwayland_surface *xsurface);
+
 static void xwm_send_focus_window(struct wlr_xwm *xwm,
 		struct wlr_xwayland_surface *xsurface) {
+	struct wlr_xwayland_surface *unfocus_surface = xwm->focus_surface;
+
+	xwm->focus_surface = xsurface;
```

Comment:
This function is a little special because it isn't a no-op when the surface is already focused (`xwm->focus_surface == xsurface`). This is necessary because we want to be able to deny `FocusIn` events. Maybe add a comment about this?